lib/cocoapods/generate/podfile_generator.rb

```diff
@@ -246,6 +250,17 @@ def swift_version
         @swift_version ||= target_definition_list.map(&:swift_version).compact.max
       end
 
+      def supported_swift_versions
+        return unless configuration.use_podfile?
+        return if target_definition_list.empty?
```

Comment:
is `target_definition_list` expensive to access?

Reply:
it's memoized